tests/test_libinteract.py

```diff
@@ -219,57 +452,136 @@ def test_generate_sc_identifiers(simulation, hc_residues_list):
+                                           fullmatrixfunc = li.calc_sc_fullmatrix, 
+                                           reslist = hc_residues_list,
+                                           mindist = False)
     assert_almost_equal(hc_mat_out, ref_hc_graph, decimal=1)
-    split_str = str_out.split("\n")[:-1]
-    for i, s in enumerate(split_str):
-        assert(s == ref_hc[i].strip())
+
+    for i, t in enumerate(table_out):
+        assert(','.join(str(x) for x in t) == ref_hc[i].strip())
 
 def test_parse_hb_file(hb_file):
     li.parse_hbs_file(hb_file)
 
 def test_do_interact_hb(simulation, hb_don_acc, ref_hb, ref_hb_graph):
     sel = 'protein'
-    str_out, hb_mat_out = li.do_hbonds(sel1 = sel,
-                                   sel2 = sel,
-                                   pdb = simulation['pdb'],
-                                   uni = simulation['uni'],
-                                   distance = 3.5,
-                                   angle = 120.0,
-                                   perco = 0.0,
-                                   perresidue = False,
-                                   do_fullmatrix = True,
-                                   other_hbs = hb_don_acc)
+    table_out, hb_mat_out = li.do_hbonds(sel1 = sel,
+                                         sel2 = sel,
+                                         pdb = simulation['pdb'],
+                                         uni = simulation['uni'],
+                                         distance = 3.5,
+                                         angle = 120.0,
+                                         perco = 0.0,
+                                         perresidue = False,
+                                         do_fullmatrix = True,
+                                         other_hbs = hb_don_acc)
 
     assert_almost_equal(hb_mat_out, ref_hb_graph, decimal=1)
-    sorted_ref_hb = sorted(ref_hb)
-    split_str = sorted(str_out.split("\n")[:-1])
-    for i, s in enumerate(split_str):
-        assert(s == sorted_ref_hb[i].strip())
+    for i, t in enumerate(table_out):
+        assert(','.join(str(x) for x in t) == ref_hb[i].strip())
+
+# check sb tables and matrix
+def test_create_table_list_sb(do_interact_sb, ref_sb_twochains, ref_sb_chains):
+   table_list = li.create_table_list(do_interact_sb['table'])
+   for i, t in enumerate(table_list[0]):
+       assert(','.join(str(x) for x in t) == ref_sb_twochains[i].strip())
+   for i, t in enumerate(table_list[1]):
+       assert(','.join(str(x) for x in t) == ref_sb_chains['intra_A'][i].strip())
+   for i, t in enumerate(table_list[2]):
+       assert(','.join(str(x) for x in t) == ref_sb_chains['inter'][i].strip())
+
+   first_table = np.sort(table_list[0], axis = 0)
+   split_tables = np.sort(np.vstack(table_list[1:]), axis = 0)
+   assert(np.array_equal(first_table, split_tables) == True)
+
+def test_create_matrix_list_sb(do_interact_sb, create_table_list_sb, 
```

Comment:
If you split arguments into multiple lines, it is better to have an argument per line (same applies to the case below).